src/lib.rs

```diff
@@ -189,6 +189,56 @@
 //! # bevy::ecs::system::assert_is_system(show_tooltips);
 //! ```
 //!
+//! Trait queries support basic change detection filtration. So to get all the components that
+//! implement the target trait, and have also changed in some way since the last tick, you can:
+//! ```ignore
+//! fn show_tooltips(
+//!     tooltips_query: Query<ChangedAll<&dyn Tooltip>>
+//!     // ...
+//! ) {
+//!     // Iterate over each entity that has tooltips, which have *changed* since the last tick
+//!     for entity_tooltips in &tooltips_query {
+//!         // Iterate over each component implementing `Tooltip` for the current entity.
+//!         for tooltip in entity_tooltips {
+//!             println!("Changed Tooltip: {}", tooltip.tooltip());
+//!         }
+//!     }
+//! }
+//! ```
+//!
+//! Similar to `ChangedAll`, we have `AddedAll`.
+//!
+//! If you know you have only one component that implements the target trait, you can use
+//! `ChangedOne` (or `AddedOne`) filters, which returns an `Option`al entity if change was detected:
+//! ```ignore
+//! fn show_tooltips(
+//!     tooltips_query: Query<ChangedOne<&dyn Tooltip>>
+//!     // ...
+//! ) {
+//!     // Iterate over each entity that has one tooltip implementing component
+//!     for maybe_changed_tooltip in &tooltips_query {
+//!         if let Some(changed_tooltip) = maybe_changed_tooltip {
+//!             println!("Changed Tooltip: {}", tooltip.tooltip());
+//!         }
+//!     }
+//! }
+//! ```
+//!
+//! or you can use `OneAddedFilter` or `OneChangedFilter` which behave more like the typical
```

Comment:
Oh, I see what you mean now. There is a big difference between these two forms.

Now that I understand the distinction, I think that it would be best to remove `AddedOne<>` and `ChangedOne<>`, in favor of just using `OneAddedFilter<>` (plus then we could remove 'Filter' from the name without confusion).

Reply:
I like that idea. And might it be useful to make a version of the filters for the `All` case too?